pkg/csi/cinder/controllerserver.go

```diff
@@ -326,6 +375,23 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 
 	name := req.Name
 	volumeID := req.GetSourceVolumeId()
+	snapshotType, snapshotTypeDeclared := req.Parameters[openstack.SnapshotType]
+
+	// Current time, used for CreatedAt
+	var ctime *timestamppb.Timestamp
+	// Size of the created snapshot, used to calculate the amount of time to wait for the backup to finish
+	// Default to 1 GB
+	snapSize := 1
+	// If true, skips creating a snapshot because a backup already exists
+	var skipSnapshot bool
+	var snap *snapshots.Snapshot
+	var backup *backups.Backup
+	var backups []backups.Backup
+
+	// Set snapshot type to 'snapshot' by default
+	if !snapshotTypeDeclared || snapshotType == "" {
```

Comment:
IMO the `snapshotTypeDeclared` is useless, especially if a snapshot type is explicitly set to an empty string. Relates to an already discussed comment below.